includes/class-alma-wc-shortcodes.php

```diff
@@ -0,0 +1,161 @@
+<?php
+/**
+ * Alma payments plugin for WooCommerce
+ *
+ * @package Alma_WooCommerce_Gateway
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	die( 'Not allowed' ); // Exit if accessed directly.
+}
+
+/**
+ * Class Wc_Alma_Shortcodes
+ */
+class Alma_Wc_Shortcodes {
```

Comment:
I feel like there could be a way to be "smarter" here with some refactoring: both `Alma_WC_Cart_Handler` and `Alma_WC_Product_Handler` inherit from the same class but have different specialized methods, `display_cart_eligibility` and `inject_payment_plan`, that perform the same thing in the end: inject the payment plan widget.

This method could have a unique name and be part of the generic handler interface, which would then allow you to have a `Alma_WC_Shortcode` (singluar) class from which you can then inherit two classes: `Alma_WC_Product_Shortcode` and `Alma_WC_Cart_Shortcode`.  
Instead of having `init_xxx_widget` methods you'd have a unique `init` method, overriden in the children classes, and anything that's common such as the `render` & `render_empty` methods can be factored in the parent class.

Reply:
I thought about that, but I consider that is not the branch subject and I prefer do this job in a specific refactoring branch